addons/mail/tests/test_mail_channel.py

```diff
@@ -197,6 +201,27 @@ def test_channel_members(self):
         self.assertEqual(channel.message_partner_ids, self.env['res.partner'])
         self.assertEqual(channel.channel_partner_ids, self.env['res.partner'])
 
+    @users('employee')
+    @mute_logger('odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_channel_chat_message_post_should_update_last_activity_time(self):
+        channel = self.env['mail.channel'].browse(self.test_channel.ids)
+        channel.write({'channel_type': 'chat'})
+        channel._action_add_members(self.partner_employee | self.partner_admin)
+        now = fields.Datetime.now()
+        with patch.object(fields.Datetime, 'now', lambda: now):
```

Comment:
Maybe explain why we need a patch, because it is not usual. Also don't you need to unpatch at the end?

Reply:
The `patch` is only effective inside the context. The meaning is to fix the return value of `Datetime.now()`. Otherwise we cannot know the exact value of `last_meaningful_action_time`. A comment is also added.